dbos/_sys_db.py

```diff
@@ -284,6 +284,7 @@ def insert_workflow_status(
         status: WorkflowStatusInternal,
         *,
         max_recovery_attempts: int = DEFAULT_MAX_RECOVERY_ATTEMPTS,
+        conn: Optional[sa.Connection] = None,
```

Comment:
Can we make this non-optional?  Fix the one place in the client where this appears so it's always used transactionally.

Reply:
Done.